### DISCUSSION TITLE
Sector creation

### DISCUSSION THREAD

**Discussion by s-merchant** [secdec-research/secdec#21] [Q&A] [answered]
Hi!  I'm trying to perform sector decomposition using the iterative_decomposition method from pySecDec.decomposition.iterative on the exponentiated forms of U and F from LoopIntegralFromPropagators.  

I combined the two polynomials into a Product.  When I try to use this Product to create a Sector, I get the error

`AssertionError: The first factor of every Product must be a monomial
`

This issue only arises upon trying to use the Product object as input for creating the Sector, not when creating the Product to begin with.

Here's what I've got: 

```from pySecDec.loop_integral import LoopIntegralFromPropagators
 from pySecDec.decomposition import Sector
 from pySecDec.decomposition.iterative import iterative_decomposition
 from pySecDec.algebra import Polynomial
 from pySecDec.algebra import Product
  
 propagators = ['k**2','(k+r)**2','(k+s)**2','(k+t)**2']
 loop_momenta = ['k']
  
 one_loop_box = LoopIntegralFromPropagators(propagators,loop_momenta,dimensionality='3-2*eps')
 
 print(one_loop_box)
 
 combined_poly = Product(one_loop_box.exponentiated_U,one_loop_box.exponentiated_F)
 print(type(one_loop_box.exponentiated_U))
 
 print("\n")
 print("printing polynomial:")
 print(combined_poly)
 
 print(combined_poly.factors[0].coeffs)
 print(len(combined_poly.factors[0].coeffs))
 
 initial_sector = Sector([combined_poly])
 
 print(initial_sector)
```

Thanks so much to anyone who has any suggestions!   I'm sure I've just overlooked something small.

**Comment by spj101** [comment, marked as answer]
Hi @s-merchant, thanks for trying our code.

Firstly, it's not super obvious what your goal is with this code. If you just want to obtain a result for this integral using sector decomposition then there is no need to understand or use the low-level algebra/decomposition functions. See the [examples](https://github.com/gudrunhe/secdec/tree/master/examples) (e.g. [elliptic2L_physical](https://github.com/gudrunhe/secdec/tree/master/examples/elliptic2L_physical)).

If you really do not want to compute this integral but instead examine some of the internal functions available within pySecDec (for example if you are following some examples in our [documentation](https://secdec.readthedocs.io/en/stable/overview.html#sector-decomposition)), then perhaps I can still help. The issue here is that `Sector` does not like products that do not have a monomial as the first factor, as the error message says. Admittedly the [documentation](https://secdec.readthedocs.io/en/v1.5.3/full_reference.html?highlight=sector#pySecDec.decomposition.Sector) is not clear about this point, we should probably improve it.

However, you can pass a list of `Polynomial` or `ExponentiatedPolynomial`, which is what we actually do in our code. 

Perhaps the following example is close to what you wanted:
```python
from pySecDec.decomposition import Sector
from pySecDec.decomposition.iterative import iterative_decomposition
from pySecDec import LoopIntegralFromPropagators

if __name__ == '__main__':
    propagators = ['k**2','(k+r)**2','(k+s)**2','(k+t)**2']
    loop_momenta = ['k']

    one_loop_box = LoopIntegralFromPropagators(propagators,loop_momenta,dimensionality='3-2*eps')

    #print('exponentiated_U', one_loop_box.exponentiated_U)
    #print('exponentiated_F', one_loop_box.exponentiated_F)

    initial_sector = Sector([one_loop_box.exponentiated_U, one_loop_box.exponentiated_F])

    decomposed_sectors = iterative_decomposition(initial_sector)

    for sector in decomposed_sectors:
        print(sector)
        print('\n')
```
You can then take the product of the `Jacobian` and `cast` polynomials in each `Sector`.

Hope this helps.
